### PR TITLE
2127 : SimpleSpannerRepository#existsById fetches all fields and inte…

### PR DESCRIPTION
**Pull request by s13o**
…rleaved child objects

### CODE REVIEW

File: spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java [hunk 1]
```diff
@@ -162,7 +162,7 @@
 	 * Deletes objects from storage in a batch.
 	 * @param objects the objects to delete from storage.
 	 */
-	void deleteAll(Iterable objects);
```
Comment (s13o) [inline spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java]: it was a warninig

File: spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java [hunk 2]
```diff
@@ -267,4 +267,13 @@
...
 			SpannerReadOptions readOptions);
+
+	/**
+	 * Returns whether an entity with the given key exists.
+	 * @param entityClass the type of the entity to check if exists.
+	 * @param key the key of the entity to check, must not be {@literal null}.
+	 * @return {@literal true} if an entity with the given key exists, {@literal false} otherwise.
+	 * @throws IllegalArgumentException if {@literal id} is {@literal null}.
+	 */
+	boolean existsById(Class entityClass, Key key);
```
Comment (s13o) [inline spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java]: a new method as it was discussed in prev PR

File: spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/event/ExecuteDmlEvent.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
 	/**
 	 * Constructor.
-	 * @param statement @param statement the DML statement which is (never {@code null})
```
Comment (s13o) [inline spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/event/ExecuteDmlEvent.java]: javadoc

File: spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/SpannerEntityProcessor.java [hunk 4]
```diff
@@ -44,7 +44,7 @@
 	<T> List<T> mapToList(ResultSet resultSet, Class<T> entityClass);
 
 	/**
-	 * Converts a set of Spanner {@link ResultSet} into a list of objects.
```
Comment (s13o) [inline spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/SpannerEntityProcessor.java]: javadoc

File: spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/ConverterAwareMappingSpannerEntityProcessor.java [hunk 5]
```diff
@@ -75,13 +75,17 @@ public ConverterAwareMappingSpannerEntityProcessor(SpannerMappingContext spanner
...
+				result.add(this.entityReader.read(entityClass,
 					resultSet.getCurrentRowAsStruct(),
 					includeColumns,
 					allowMissingColumns));
+			}
+		}
+		finally {
+			resultSet.close();
 		}
-		resultSet.close();
```
Comment (s13o) [inline spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/ConverterAwareMappingSpannerEntityProcessor.java]: better to close in "finally"

File: spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java [hunk 6]
```diff
@@ -161,8 +162,8 @@ private SpannerStatementQueryExecutor() {
 	}
 
 	/**
-	 * Gets a query that returns the rows associated with a parent entity. This function is
```
Comment (s13o) [inline spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java]: javadoc

**Comment by s13o** [comment]
Can someone explain to me what has happened with the overage report?
I can hardly understand what caused this delta

**Comment by elefeint** [comment]
The coverage delta is due to integration tests not running on PRs; it is largely expected on Spanner PRs.

**Comment by meltsufin** [comment]
@s13o It looks like this PR is making some additional improvements to the code aside from avoiding the fetching of all fields. Once we merge #2136, maybe we can rebase this PR and try to get in those other enhancements you did here.

**Comment by meltsufin** [comment]
Superseded by #2136,

**Pull request closed without merging**